src/Symfony/Component/Form/FormFactory.php

```diff
@@ -126,7 +126,7 @@ public function createBuilderForProperty($class, $property, $data = null, array
 
         // user options may override guessed options
         if ($typeGuess) {
-            $options = array_merge($typeGuess->getOptions(), $options);
+            $options = array_replace_recursive($typeGuess->getOptions(), $options);
```

Comment:
@xabbuh I think it's subjective, we might want this situation for `attr` option some times, but the current implementation (e.g. `$pattern` & `$maxLength`) vote for `array_replace_recursive()`, so I don't see why not be consistent with other options ;) 

But your decide, do you think we should do this just for the `attr` option?